runtime/src/lib.rs

```diff
@@ -272,15 +273,14 @@ pub trait Listener: Sync + Send + 'static {
 /// messages over a network connection.
 pub trait Sink: Sync + Send + 'static {
     /// Send a message to the sink.
-    fn send(&mut self, msg: &[u8]) -> impl Future<Output = Result<(), Error>> + Send;
+    fn send(&mut self, msg: Bytes) -> impl Future<Output = Result<(), Error>> + Send;
```

Comment:
I think `Bytes` (what you have) makes sense here. We are very commonly doing some sort of broadcast where the caller provides `Bytes` as input.

On the receive side, I think we could get away with `Vec<u8>` but ok with keeping it consistent on send/recv. I think for storage, we can do always `Vec<u8>`.